intellij-plugin-verifier/verifier-intellij/src/main/java/com/jetbrains/pluginverifier/usages/javaPlugin/UndeclaredDependencyOnJavaPluginProblem.kt

```diff
@@ -1,26 +1,22 @@
-/*
- * Copyright 2000-2020 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
- */
-
-package com.jetbrains.pluginverifier.warnings
+package com.jetbrains.pluginverifier.usages.javaPlugin
 
 import com.jetbrains.pluginverifier.results.presentation.ClassGenericsSignatureOption
 import com.jetbrains.pluginverifier.results.presentation.ClassOption
 import com.jetbrains.pluginverifier.results.presentation.formatClassLocation
+import com.jetbrains.pluginverifier.results.problems.CompatibilityProblem
 import com.jetbrains.pluginverifier.usages.formatUsageLocation
-import com.jetbrains.pluginverifier.usages.javaPlugin.JavaPluginClassUsage
+import java.util.*
 
-data class NoExplicitDependencyOnJavaPluginWarning(
```

Comment:
Why don't still use `data class` here?

Maybe we can make all problems `data class`es (but in a separate branch of course). What do you think?

Reply:
Descendants of `CompatibilityProblem` are not declared as data classes anywhere. This is to be consistent.

On the other hand, _compatibility problems_ just hold data, so they are a good candidates. Let's do that on a separate branch.